### PR TITLE
fix(developer): support Windows and Unicode names in .ttf

### PR DESCRIPTION
**Pull request by mcdurdin**
The ttfmeta library is extremely rudimentary. It does next to no error checking, and supports only a small part of the TTF spec. It did not support NAME platforms 0 or 3 (the most common platform, Windows), and these are encoded in UTF-16BE. This patch adds support for those platforms and throws if an unsupported platform/encoding is encountered.

Now, the ttfmeta library is unsupported and gone (#11630). It is based on another library, ttfinfo, last updated 11 years ago. This is not great. But the other Node libraries that I located introduce further dependencies or complex builds (e.g. WASM) and that's not great either, for a function used only in one place in our build environment.

https://learn.microsoft.com/en-us/typography/opentype/spec/name

The test-font-family.ts runs a basic test against Lateef font, and has a skipped test that can be run (with minor edits) against all fonts in the keyboard repository. I have done this to verify that there are no fonts that return errors, but have disabled this test for deployment for now.

Fixes: #11625

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-keyboard-info/test/test-font-family.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      console.log(`Testing ${file}`);
+      const fontData: Uint8Array = fs.readFileSync(file);
+      const facename = await getFontFamily(fontData);
+      assert.isNotEmpty(facename);
+      assert.isFalse(facename.includes('\u0000'));
+    }
+
+    // To enable this, we need to have access to the shared fonts in the
+    // keyboards repo
+    await testFonts('.../keyboards/release/shared/fonts/');
```
Comment (darcywong00) [inline developer/src/kmc-keyboard-info/test/test-font-family.ts:48]: Maybe too much for now.

If this was a separate npm utility package, then could the keyboards repo use it to test the fonts on CI builds?

**Review by darcywong00 — APPROVED** [review]
lgtm

File: developer/src/kmc-keyboard-info/test/test-font-family.ts (hunk 1 above)
Comment (mcdurdin) [reply developer/src/kmc-keyboard-info/test/test-font-family.ts:48]: future issue #11641

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.190-beta](https://keyman.com/downloads/releases/beta/18.0.190)

**Pull request merged**